DashKit/DashPumpManager.swift

```diff
@@ -625,22 +623,31 @@ open class DashPumpManager: PumpManager {
                 completion(error)
             case .success(let status):
                 self.mutateState({ (state) in
-                    state.expirationReminderDate = state.podActivatedAt?.addingTimeInterval(Pod.lifetime - intervalBeforeExpiration)
+                    state.scheduledExpirationReminderOffset = intervalBeforeExpiration
                     state.updateFromPodStatus(status: status)
                 })
                 completion(nil)
             }
         }
     }
     
-    public var allowedExpirationReminderDateRange: ClosedRange<Date>? {
+    public var allowedExpirationReminderDates: [Date]? {
         guard let expiration = podExpiresAt else {
             return nil
         }
         
-        let earliest = expiration.addingTimeInterval(.hours(-24))
-        let latest = expiration.addingTimeInterval(.hours(-1))
-        return earliest...latest
+        let allDates = Array(stride(from: 24, through: 1, by: -1)).map { (i: Int) -> Date in
```

Comment:
[nit] Consider `from: -24, through: -1, by: 1` and getting rid of the `-i` on the line below?

Reply:
Seems fairly equivalent to me, but it sounds like you think it reads better that way, so I'll update it.